src/odb/src/db/dbTechLayer.h

```diff
@@ -85,6 +85,7 @@ struct dbTechLayerFlags
   uint has_protrusion_ : 1;
   uint has_alias_ : 1;
   uint has_xy_pitch_ : 1;
+  uint has_first_last_pitch_ : 1;
```

Comment:
This increases the number of bits in the bitfield struct to 33. in the stream functions, the flags are handled as uint32_t. I recommend removing the flag and depending on the value of first_last_patch_ variable. If it's greater than or equal to 0, then the layer has a first last pitch.
If you prefer to use the bool flag, then make sure to correctly read/write and the tech layer flags. Also move the has_first_last_pitch_ to the end of the struct so that when you read an older db, it doesn't mess up the variables. And for backward compatibility, you will need to handle older dbs with a 32-bit flag, and new dbs with a 64-bit flag.